### PR TITLE
Make alias not behave as a keyword outside of enaml blocks

### PR DESCRIPTION
**Pull request by frmdstryr**
For #380

### CODE REVIEW

File: enaml/core/parser/base_parser.py [hunk 1]
```diff
@@ -497,29 +496,23 @@ def p_pragma_arg3(self, p):
     # -------------------------------------------------------------------------
     # AliasExpr
     # -------------------------------------------------------------------------
-    def p_alias_expr1(self, p):
+    def d_alias_expr1(self, p):
         ''' alias_expr : ALIAS NAME NEWLINE '''
```
Comment (sccolbert) [inline enaml/core/parser/base_parser.py]: why are you using d_ instead of p_?

File: enaml/core/parser/base_parser.py [hunk 2]
```diff
@@ -439,7 +439,6 @@ def p_enamldef_suite_item(self, p):
     def p_enamldef_simple_item1(self, p):
         ''' enamldef_simple_item : binding
                                  | ex_binding
-                                 | alias_expr
```
Comment (sccolbert) [inline enaml/core/parser/base_parser.py]: ??

File: enaml/core/parser/base_parser.py [hunk 3]
```diff
@@ -497,29 +496,23 @@ def p_pragma_arg3(self, p):
...
         node = enaml_ast.AliasExpr()
         node.lineno = p.lineno(1)
         node.name = p[2]
         node.target = p[2]
         p[0] = node
 
-    def p_alias_expr2(self, p):
-        ''' alias_expr : ALIAS NAME COLON NAME NEWLINE '''
+    def d_alias_expr2(self, p):
+        ''' alias_expr : ALIAS NAME COLON dotted_name NEWLINE '''
```
Comment (sccolbert) [inline enaml/core/parser/base_parser.py]: why are you removing support for `NAME`? You can alias something that isn't a dotted name.

**Comment by frmdstryr** [comment]
When I remove `alias` from the reserved dict I get this error (after re-enabling logging):

```
ERROR: /home/user/enaml/enaml/core/parser/base_parser.py:501: Symbol 'ALIAS' used, but not defined as a token or a rule
ERROR: /home/user/enaml/enaml/core/parser/base_parser.py:509: Symbol 'ALIAS' used, but not defined as a token or a rule
ERROR: /home/user/enaml/enaml/core/parser/base_parser.py:517: Symbol 'ALIAS' used, but not defined as a token or a rule
WARNING: Token 'STRING_CONTINUE' defined, but not used
WARNING: Token 'STRING_END' defined, but not used
WARNING: Token 'STRING_START_SINGLE' defined, but not used
WARNING: Token 'STRING_START_TRIPLE' defined, but not used
WARNING: Token 'WS' defined, but not used
WARNING: There are 5 unused tokens
ERROR: Infinite recursion detected for symbol 'alias_expr'
Traceback (most recent call last):
  File "/home/user/miniconda3/envs/enaml3/bin/enaml-run", line 11, in <module>
    load_entry_point('enaml', 'console_scripts', 'enaml-run')()
  File "/home/user/miniconda3/envs/enaml3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/user/miniconda3/envs/enaml3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2843, in load_entry_point
    return ep.load()
  File "/home/user/miniconda3/envs/enaml3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2434, in load
    return self.resolve()
  File "/home/user/miniconda3/envs/enaml3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2440, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/user/enaml/enaml/runner.py", line 20, in <module>
    from enaml.core.parser import parse
  File "/home/user/enaml/enaml/core/parser/__init__.py", line 30, in <module>
    _parser = Python36EnamlParser()
  File "/home/user/enaml/enaml/core/parser/base_parser.py", line 171, in __init__
    debug=0,
  File "/home/user/miniconda3/envs/enaml3/lib/python3.7/site-packages/ply-3.11-py3.7.egg/ply/yacc.py", line 3432, in yacc
    raise YaccError('Unable to build parser')
ply.yacc.YaccError: Unable to build parser
```

If I remove alias from reserved then change the `alias_expr` to use NAME instead of ALIAS then it becomes ambiguous with the `storage_expr` and attempts to make attr expressions alias's.

So I'm not sure how to make it select the parsing rule without doing it manually in code.

**Comment by sccolbert** [comment]
I don’t have time to deal with this right now. Since there’s a variety of workarounds, it’s not a high priority thing. There should be a way to have PLY deal with keywords only inside a certain block.

**Comment by frmdstryr** [comment]
No worries, I'll re-open if I find a cleaner way to do it.

**Pull request closed without merging**